spec/factory_specs/course_spec.rb

```diff
@@ -30,4 +30,23 @@
       expect(CourseEnrichment.where("created_at < ?", 4.days.ago).count).to eq(1)
     end
   end
+
+  context "unpublished_with_primary_maths" do
+  subject! { TestSetup.unpublished_course_with_primary_maths }
+
+    its(:name) { should eq("unpublished_with_primary_maths course name")}
+
+    it "has the correct number of associations" do
+      expect(Site.count).to eq(1)
+      expect(Provider.count).to eq(1)
+      expect(Organisation.count).to eq(1)
+      expect(User.count).to eq(1)
```

Comment:
When we add `TestSetup` to the `before :all`, this fail. Probably better to change `subject!` to `subject` and do:

```
      expect { subject }.not_to change { Site.count }
```

... etc

Reply:
Also, I think the factory we're testing here should only create one `Site` and `Provider`, the `provider` factory should only create one `Organisation`, and the `organisation` factory should only create one `User`.